[BUG 3.5.0 beta] Fix scrolling inside bootstrap modals (See New PR #9140)

cyrez opened the pull request:
Fix scrolling inside bootstrap modals (small height of window viewport: mobile, window resized...).
Corrects the media modal padding on small devices.
#### Steps to reproduce the issue

**Isis template**
In admin test new bootstrap modals introduced in 3.5.0 beta2 : 
- Add New Menu Item : **Menu Item Type** (component views) or **Link Image** (image) option
- Add New Article : **Created By** (users), **Intro Image** (image)
- In list of items : test **Batch** modals
- Everywhere a bootstrap modal is used...

**Protostar template**
In frontend : 
- Edit Article : **Versions**, **Intro Image** (image) ...
- Everywhere a bootstrap modal is used...
#### Expected result

We should be able to scroll inside the modal, and use all options when using administration on a mobile or small screen device.
#### Actual result in 3.5.0 beta2

You may not be able to scroll the modal on iOS (iPhone) (tests on other mobile OS are welcome!)
On window resized (small height), modal could be off screen (bottom content not accessible)

BEFORE PATCH on iPhone 4S iOS 9.2.1
![iphone4s-before-patch](https://cloud.githubusercontent.com/assets/2385058/13099219/6dd11f00-d530-11e5-811f-9cd734b556c0.jpg)
_Modal off screen. Not possible to scroll._
#### Additional comments

Scrolling not working on mobile is a known issue of Bootstrap modals, not fixed in BS 2 and 3 : https://github.com/twbs/bootstrap/issues/14839 (and maybe related to a webkit bug: https://bugs.webkit.org/show_bug.cgi?id=153856)
To set a max-height for modal-body class depending on the current viewport height solves the issue of not being able to view bottom of the modal window. (resize your window to arround 300px height before and after patch)
#### Testing

Apply the patch, and you may be able to scroll inside all bootstrap modals.

AFTER PATCH on iPhone 4S iOS 9.2.1
![iphone4s-after-patch](https://cloud.githubusercontent.com/assets/2385058/13099236/9611b5ba-d530-11e5-82ed-e22ef34f85f3.jpg)
_Modal adapts to screen. Scroll is working._

Note: Review of the comments by a native English speaker will be appreciated ;-)

cyrez committed:
Fix iOS scrolling inside bootstrap modals

Fix iOS scrolling inside bootstrap modals.
Corrects the media modal padding on small devices.

cyrez committed:
Fix iOS scrolling inside bootstrap modals

Fix iOS scrolling inside bootstrap modals.
Corrects the media modal padding on small devices.

cyrez committed:
Fix iOS scrolling inside bootstrap modals

Set the max-height of modal-body depending on the window viewport height.
Fix small screens scrolling inside bootstrap modals.

cyrez committed:
Revert Fix small screens scrolling inside bootstrap modals. MOVED to layout/modal/main.php

JoomliC commented:
Thanks to first feedback from @dgt41 ( :+1: ) i will update tomorrow this PR with protostar css fix, and more testing instruction with screen shots before/after patch.

cyrez committed:
Fix scrolling inside bootstrap modals (small window viewport height)

Add fix to Protostar

cyrez committed:
Add comments and max-height for iframe

 Fix Scrolling inside bootstrap modals on small screens (adapt to window viewport and avoid modal off screen).

cyrez committed:
Add comments and max-height for iframe

Fix scrolling inside bootstrap modals on small screens (adapt to window viewport and avoid modal off screen).

cyrez committed:
Missing empty end line

JoomliC commented:
PR now updated for Protostar, and ready for testing! ;-)

cyrez committed:
Merge branch 'joomla/staging' into Fix-iOS-scrolling-inside-bootstrap-modals

# Conflicts:
#	templates/protostar/css/template.css
#	templates/protostar/less/template.less

cyrez committed:
Revert "Merge branch 'joomla/staging' into Fix-iOS-scrolling-inside-bootstrap-modals"

This reverts commit 79970da0a1658e67292e43922b00f8a16007e79f.

cyrez committed:
update to staging

cyrez committed:
Revert "update to staging"

This reverts commit 3cea7b4a570edc70767d8f46814a7dd395838c81.

JoomliC commented:
Closed this PR for another one, with no conflict with the base branch : https://github.com/joomla/joomla-cms/pull/9140
